lib/views/shared/settings.dart

```diff
@@ -41,16 +41,30 @@ abstract class SettingsShared<T extends SettingsState> extends State<T> {
     }
   }
 
+  canWriteToDirectory(String result) async {
```

Comment:
This should return a boolean, the try block at the call site can be moved here instead.